src/psyclone/psyir/frontend/fortran.py

```diff
@@ -60,12 +64,37 @@ def psyir_from_source(self, source_code):
 
         :returns: PSyIR representing the provided Fortran source code.
         :rtype: :py:class:`psyclone.psyir.nodes.Node`
+
         '''
         string_reader = FortranStringReader(source_code)
         parse_tree = self._parser(string_reader)
         psyir = self._processor.generate_psyir(parse_tree)
         return psyir
 
+    def psyir_from_literal_expression(self, source_code):
```

Comment:
I see this accept inputs like "-3.0 + 1.0" and "ABS(-3.0)". Isn't this pushing the semantics of "literal"? But tbh I don't have a good name alternative. Maybe "psyir_from_non_symbolic_expression", but I am not sure.

Maybe replacing the "literal expression" from the docstring and explaining the subtleties of what this method accepts or not could be enough.

Reply:
I meant that it was an expression involving only literals. I was imagining that it might be useful to extend it in the future to support expressions involving symbols (we could pass it a symbol table) so I could rename it to `psyir_from_expression` with a note to say that symbols are not currently supported?

Reply:
Ah yes, if this is the intent the `psyir_from_expression` is a nicer name, and I am fine the the note saying symbols are not supported for now.

Reply:
Done.